batchlib/outliers/outlier.py

```diff
@@ -10,28 +10,20 @@
 
 
 def get_outlier_predicate(config):
-    if hasattr(config, 'outliers_dir') and config.outliers_dir is not None:
-        outliers_dir = config.outliers_dir
+    if hasattr(config, 'misc_folder') and config.misc_folder is not None:
+        outliers_dir = os.path.join(config.misc_folder, 'tagged_outliers')
     else:
         outliers_dir = DEFAULT_OUTLIER_DIR
 
-    if hasattr(config, 'plate_name') and config.plate_name is not None:
-        plate_name = config.plate_name
-        if isinstance(plate_name, list):
-            assert len(plate_name) == 1, \
-                f'Only a single plate name is allowed, but {len(plate_name)} were given to the outlier predicate'
-            plate_name = plate_name[0]
-        logger.info(f'Using plate name: {plate_name}')
+    logger.info(f"Trying to parse 'plate_name' from the input folder: {config.input_folder}")
+    plate_name = plate_name_from_input_folder(config.input_folder)
+    if plate_name is not None:
+        logger.info(f"plate_name found: {plate_name}")
     else:
-        logger.info(f"Trying to parse 'plate_name' from the input folder: {config.input_folder}")
-        plate_name = plate_name_from_input_folder(config.input_folder)
-        if plate_name is not None:
-            logger.info(f"plate_name found: {plate_name}")
-        else:
-            logger.info(f"Cannot parse plate_name. Outlier detection will be skipped")
-            # no plate name was given and it cannot be parsed from the config.input_folder
-            # so we skip outlier detection, i.e. assign outlier: None to each of the images
-            return lambda im: None
+        logger.info(f"Cannot parse plate_name. Outlier detection will be skipped")
+        # no plate name was given and it cannot be parsed from the config.input_folder
+        # so we skip outlier detection, i.e. assign outlier: -1 to each of the images
+        return lambda im: -1
```

Comment:
This changes the semantics of the `OutlierPredicate.__call__` method:
```
Returns:
            True if the img_file was marked as an outlier
            False if the img_file was marked as a valid image
            None if the img_file was skipped during tagging,
            or outliers are not available for a given plate (no outlier CSV file)
```
it should be `return lambda im: None`

Reply:
Are you sure? At least here, I am assuming `-1` if we don't have the outlier table:
https://github.com/hci-unihd/batchlib/blob/workflows/batchlib/analysis/cell_level_analysis.py#L496-L500

Reply:
yes, so please have a look at the implementation: https://github.com/hci-unihd/batchlib/blob/master/batchlib/outliers/outlier.py#L99-L104. Basically the CSV files contain 1, 0 and -1, which is then mapped to `True`, `False`, `None` for convenience. If we just want to keep the numeric values I'm fine with that, but we need to make sure it's consistent in the code

Reply:
I see.
Yes, numeric values are actually more convenient in the downstream code, because we need to save the table to hdf5 and get rid of `Nones` to properly serialize this.
I would just remove the `True`, `False`, `None` conversion then.

Reply:
yes, let's do that then. But please have a look at other places in the code, e.g. this needs to be changed as well: https://github.com/hci-unihd/batchlib/blob/master/batchlib/analysis/summary.py#L162. I can go over the code later when this is merged and make sure that we are consistent